src/main/java/impl/image/RgbConvolutionProvider.java

```diff
@@ -0,0 +1,42 @@
+package impl.image;
+
+import api.image.ConvolutionProvider;
+import java.awt.*;
+
+public class RgbConvolutionProvider implements ConvolutionProvider {
+    @Override
+    public Color[][] apply(Color[][] image, double[][] kernel) {
+        var imageRows = image.length;
+        var imageColumns = image[0].length;
```

Comment:
Isn't it possible `ArrayIndexOutOfBoundsException`?

Reply:
Yes, it is possible. But I'll begin to pay attention to such errors with the second homework